include/revng/PipelineC/Prototypes.h

```diff
@@ -316,12 +383,12 @@ void rp_target_destroy(rp_target *target);
 rp_kind *rp_target_get_kind(rp_target *target);
 
 /**
- * Serializes target into a string, target cannot be NULL.
+ * Serializes target into a string.
  *
  * \note The return string is owned by the caller. Destroy with
  *       rp_string_destroy.
  */
-char *rp_target_create_serialized_string(rp_target *target);
+char * /*owning*/ rp_target_create_serialized_string(rp_target *target);
```

Comment:
Is this passing `check-conventions`? I'd add a space between the comment delimiters.

Reply:
@mrjackv which format do you want for the string `/*onwing*/` ?

Reply:
> Is this passing check-conventions?
Yes
> which format do you want for the string /*onwing*/ ?
doesn't matter whitespace-wise

Reply:
The comment solution is not so exciting after all, one day we could 1) introduce a custom attribute/annotation, 2) write a clang-based tool that generated a metadata file where it prints all the functions that return an owning pointer.

But at this stage, it seems overkill, so let's go with what we have.